### DISCUSSION TITLE
Config files: Agents and Tasks

### DISCUSSION THREAD

**Discussion by smortezah** [crewAIInc/crewAI#1739] [General]
Hi,

I'm wondering what the motivation for separating agents.yaml and tasks.yaml from the source code is? Because YAML is based on text and raises issuses such as the following:
- When an agents name changes in the source python code, we have to manually alter it in the agents.yaml file. The same happens with the tasks.config.
- Also, when we use for example tool names in the backstories, and later we change the names of those tools, again we have to change everything manually in the separate YAML files.

Isn't it better to change the @CrewBase in a way that it reads from the source Agent and Task definitions in the python code, rather than reading configs from YAML files?

**Comment by manni07** [comment]
From my understanding you can just use the source code deleting the yaml files. Did you try it?
